lib/SILOptimizer/Mandatory/Differentiation.cpp

```diff
@@ -4380,11 +4386,17 @@ class AdjointEmitter final : public SILInstructionVisitor<AdjointEmitter> {
       if (activeBBValues.empty())
         continue;
       // Otherwise, if the original block has active values:
-      // - For each active value, add adjoint value arguments to the adjoint
-      //   block.
+      // - For each active buffer in the original block, allocate a new local
+      //   buffer in the adjoint entry. (All adjoint buffers are allocated in
+      //   the adjoint entry and deallocated in the adjoint exit.)
+      // - For each active value in the original block, add adjoint value
+      //   arguments to the adjoint block.
       for (auto activeValue : activeBBValues) {
         if (activeValue->getType().isAddress()) {
-#error "FIXME: create local allocations"
+          // Allocate and zero initialize a new local buffer using
+          // `getAdjointBuffer`.
+          builder.setInsertionPoint(adjoint.getEntryBlock());
```

Comment:
`builder.setInsertionPoint(adjoint.getEntryBlock());` is crucial because `getAdjointBuffer` currently assumes that `builder` is at a valid insertion point before `emitZeroIndirect`.